e2e_test/batch/describe_fragments.slt

```diff
@@ -0,0 +1,241 @@
+# Test DESCRIBE FRAGMENTS with complex relations and schemas
+
+# Setup test schema
+statement ok
+CREATE SCHEMA describe_plan_test;
+
+statement ok
+CREATE TABLE describe_plan_test.tbl (
+    id INT PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    age INT,
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+    data JSONB
+);
+
+# Test materialized view with complex query
+statement ok
+CREATE MATERIALIZED VIEW describe_plan_test.complex_mv AS
+SELECT
+    t.name,
+    COUNT(*) as count,
+    AVG(t.age) as avg_age,
+    jsonb_agg(t.data) as all_data
+FROM describe_plan_test.tbl t
+WHERE t.age > 18
+GROUP BY t.name
+HAVING COUNT(*) > 1;
+
+# Test index with included columns and custom distribution
+statement ok
+CREATE INDEX idx ON describe_plan_test.tbl (name DESC, age)
+INCLUDE (created_at)
+DISTRIBUTED BY (name);
+
+# Test source with complex schema and options
+statement ok
+CREATE SOURCE describe_plan_test.src (
+    id INT,
+    event_time TIMESTAMP,
+    payload STRUCT<name VARCHAR, value INT>,
+    metadata MAP(VARCHAR, VARCHAR)
+) WITH (
+    connector = 'datagen',
+);
+
+# Test sink with transformation
+statement ok
+CREATE SINK describe_plan_test.snk AS
+SELECT
+    id,
+    name,
+    age,
+    created_at,
+    data->>'type' as type
+FROM describe_plan_test.tbl
+WHERE age >= 21
+WITH (connector ='blackhole')
+
+# Test view with subquery and joins
+statement ok
+CREATE VIEW describe_plan_test.complex_view AS
+WITH aged_users AS (
+    SELECT name, age
+    FROM describe_plan_test.tbl
+    WHERE age > 25
+)
+SELECT
+    t.name,
+    t.age,
+    m.count as appearance_count
+FROM aged_users t
+LEFT JOIN describe_plan_test.complex_mv m ON t.name = m.name;
+
+# Test DESCRIBE FRAGMENTS for each relation
+# Use psql_validate.py since the fragment id is not deterministic
+
+system ok
+psql_validate.py --sql "DESCRIBE FRAGMENTS describe_plan_test.tbl" \
```

Comment:
A bit sad to see system command being abused in this way. 😢 Perhaps we can introduce some sort of non-exact matching in SLT.

Reply:
Yeah. But it needs a bit design and iterate.. It's harder than I imagined and I spent >1h to debug yesterday. So prefer to keep it here to iterate faster.

Reply:
e.g., one difficulty is the error message when match failed..

Reply:
🤔 Actually, I'm thinking stuff like PLAN is already not suitable for (original) SLT.

Reply:
In the meantime, maybe we don't want SLT any more https://github.com/risingwavelabs/risingwave/issues/20525#issuecomment-2811901092